### DISCUSSION TITLE
Help when move to root page again

### DISCUSSION THREAD

**Discussion by DanielSCBO** [PrismLibrary/Prism#3028] [.NET MAUI] [answered]
Hello i have a app, and im some cases got this error

```
[DOTNET] Prism.Ioc.ContainerResolutionException: An unexpected error occurred while resolving 'Prism.Navigation.INavigationRegistry'
[DOTNET]  ---> System.NullReferenceException: Object reference not set to an instance of an object.
[DOTNET]    at DryIoc.Resolver.Resolve(IResolver resolver, Type serviceType, Object[] args, IfUnresolved ifUnresolved, Type requiredServiceType, Object serviceKey) in /_/src/DryIoc/Container.cs:line 8652
[DOTNET]    at Prism.Container.DryIoc.DryIocScopedProvider.Resolve(Type type, ValueTuple`2[] parameters) in /_/src/Prism.Container.DryIoc/DryIocScopedProvider.cs:line 54
[DOTNET]    --- End of inner exception stack trace ---
[DOTNET]    at Prism.Container.DryIoc.DryIocScopedProvider.Resolve(Type type, ValueTuple`2[] parameters) in /_/src/Prism.Container.DryIoc/DryIocScopedProvider.cs:line 58
[DOTNET]    at Prism.Container.DryIoc.DryIocScopedProvider.Resolve(Type type) in /_/src/Prism.Container.DryIoc/DryIocScopedProvider.cs:line 41
[DOTNET]    at Prism.Ioc.IContainerProviderExtensions.Resolve[INavigationRegistry](IContainerProvider provider) in /_/src/Prism.Container.Abstractions/IContainerProviderExtensions.cs:line 16
[DOTNET]    at Prism.Navigation.PageNavigationService.get_Registry() in D:\a\Prism\Prism\src\Maui\Prism.Maui\Navigation\PageNavigationService.cs:line 44
[DOTNET]    at Prism.Navigation.PageNavigationService.ProcessNavigationForContentPage(Page currentPage, String nextSegment, Queue`1 segments, INavigationParameters parameters, Nullable`1 useModalNavigation, Boolean animated) in D:\a\Prism\Prism\src\Maui\Prism.Maui\Navigation\PageNavigationService.cs:line 481
[DOTNET]    at Prism.Navigation.PageNavigationService.ProcessNavigation(Page currentPage, Queue`1 segments, INavigationParameters parameters, Nullable`1 useModalNavigation, Boolean animated) in D:\a\Prism\Prism\src\Maui\Prism.Maui\Navigation\PageNavigationService.cs:line 364
[DOTNET]    at Prism.Navigation.PageNavigationService.NavigateAsync(Uri uri, INavigationParameters parameters) in D:\a\Prism\Prism\src\Maui\Prism.Maui\Navigation\PageNavigationService.cs:line 305
[DOTNET] System.NullReferenceException: Object reference not set to an instance of an object.
[DOTNET]    at DryIoc.Resolver.Resolve(IResolver resolver, Type serviceType, Object[] args, IfUnresolved ifUnresolved, Type requiredServiceType, Object serviceKey) in /_/src/DryIoc/Container.cs:line 8652
[DOTNET]    at Prism.Container.DryIoc.DryIocScopedProvider.Resolve(Type type, ValueTuple`2[] parameters) in /_/src/Prism.Container.DryIoc/DryIocScopedProvider.cs:line 54
```

The error only happens when I page **nav/Login** open the page **Settings** then I go back and go to the page **../MainPage** and if on the main page I logout and go back to **nav/Login** I get the error, If don't open the settings it works correctly

I already tried to simulate it in another project but I didn't get error, it work correctly, any tips?

Thanks

**Comment by DanielSCBO** [comment]
Try to debug prism, 

When nav works my _container and Resolved, when dont work Resolved is null.

![image](https://github.com/PrismLibrary/Prism/assets/5363187/bee75501-449d-484b-8d63-185df54be728)

I use Modules it will be because of this

**Comment by dansiegel** [comment]
Without a reproduction of the issue it's impossible for anyone to tell you what the problem you're facing is. If you are using Regions and you attempt to Navigate to a Page that has a Region and the Page has not been properly disposed you will see an error similar to this.

**Comment by DanielSCBO** [comment]
Sorry for late reply, the problem is when i use Modules, and navigate from there. I converted another project to .NET 8 and I'm having this problem again. I'm trying to make an example project

**Comment by DanielSCBO** [comment]
I discovered the problem

I have several classes registered with Singleton INavigationService  If the Singleton initialization is in the module this happens.
I passed the class to scoped and it worked

**Reply by dansiegel** [reply, marked as answer]
The INavigationService has never in any version supported being referenced from a singleton. Navigation in Xamarin.Forms/MAUI is based on the page you are Navigating from. Without a reference page for the Navigation Service any Navigation would be absolute.

**Reply by DanielSCBO** [reply]
After debugging inside the Prism library I understood this. It was a minor thing that gave me headaches

Thanks for help
